dependency/vault_read.go

```diff
@@ -176,7 +176,20 @@ func (d *VaultReadQuery) readSecret(clients *ClientSet) (*api.Secret, error) {
 func deletedKVv2(s *api.Secret) bool {
 	switch md := s.Data["metadata"].(type) {
 	case map[string]interface{}:
-		return md["deletion_time"] != ""
+		deletionTime, ok := md["deletion_time"].(string)
+		if !ok {
+			// Not a string, end early
```

Comment:
```suggestion
			// key not present, end early
```

Reply:
Since we're casting to a string, we'll only go down the `!ok` codepath will only be gone down if the key isn't present or the value isn't a string. I'll improve the comment to show that it could be both :)

Reply:
Updated! Please let me know if there's anything else you'd need before approving :)